### PR TITLE
fix(cli): Overridding name/generateName when creating CronWorkflows if specified

### PR DESCRIPTION
**Pull request by terrytangyuan**
Currently we only copy the workflow spec after we applied the submit options but generateName/name cannot be overridden. This PR fixes this issue.

Signed-off-by: Yuan Tang <terrytangyuan@gmail.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

Tips:

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it does not need to pass
* Sign-off your commits to pass the DCO check: `git commit --signoff`.
* Run `make pre-commit -B` to fix codegen or lint problems. 
* Say how how you tested your changes. If you changed the UI, attach screenshots.

### CODE REVIEW

**Review by simster7 — CHANGES_REQUESTED** [review]
The name of the workflow is derived from the name of the cron workflow. Can you elaborate on how the CLI name override would work with cron workflows?

**Comment by terrytangyuan** [comment]
The reason why this PR is necessary is because `--name` and `--generate-name` are not effective.

The current code is not very intuitive but here is how this works:
1. `ApplySubmitOpts` takes `--name` and `--generate-name` to add them to a workflow spec.
1. We copy these metadata into CronWorkflow's metadata since previously we only copied the `newWf.Spec` into `cronWf.Spec.WorkflowSpec` without the metadata. 

These metadata names are only used in CronWorkflow. The created CronWorkflow will determine the names of its child workflows.

File: cmd/argo/commands/cron/create.go [hunk 1]
```diff
@@ -81,6 +81,14 @@ func CreateCronWorkflows(filePaths []string, cliOpts *cliCreateOpts, submitOpts
 			log.Fatal(err)
 		}
 		cronWf.Spec.WorkflowSpec = newWf.Spec
```
Comment (alexec) [inline cmd/argo/commands/cron/create.go:83]: This seems hacky,  it your comment suggests there is a good reason. Maybe add a code comment here?

**Review by alexec — CHANGES_REQUESTED** [review]

File: cmd/argo/commands/cron/create.go (hunk 1 above)
Comment (terrytangyuan) [reply cmd/argo/commands/cron/create.go:83]: Thanks. Added comment.

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
@simster7 to approve before auto-merge.

**Comment by sarabala1979** [comment]
LGTM, @simster7 Can you quick look this PR and approve it?

**Review by simster7 — APPROVED** [review]
Sorry for delay. LGTM

**Pull request merged**